src/main/java/org/myrobotlab/service/WebkitSpeechRecognition.java

```diff
@@ -17,276 +12,23 @@
  *
  */
 public class WebkitSpeechRecognition extends AbstractSpeechRecognizer {
-  public HashMap<String, String> languagesList = new HashMap<String, String>();
-  // used for angular
-  public String currentWebkitLanguage;
 
+  private static final long serialVersionUID = 1L;
+  
   /**
-   * TODO: make it's own class. TODO: merge this data structure with the
-   * programab oob stuff?
-   *
+   * mic image
    */
-  public class Command {
-    public String name;
-    public String method;
-    public Object[] params;
-
-    Command(String name, String method, Object[] params) {
-      this.name = name;
-      this.method = method;
-      this.params = params;
-    }
-  }
-
+  String img = "../WebkitSpeechRecognition/mic.png";
+  
   /**
-   * 
+   * if currently running chrome
    */
-  private static final long serialVersionUID = 1L;
-  public String lastThingRecognized = "";
-
-  boolean autoListen;
-  boolean continuous;
-
-  HashMap<String, Command> commands = new HashMap<String, Command>();
-
-  private boolean listening = false;
-  private boolean speaking = false;
-
-  private long lastAutoListenEvent = System.currentTimeMillis();
-  public boolean stripAccents = false;
-  private boolean lockOutAllGrammar = false;
-  private String lockPhrase = "";
-
-  public WebkitSpeechRecognition(String n, String id) {
-    super(n, id);
-    languagesList.put("en-US", "English - United States");

```

Comment:
how will you get the list of supported languages from webkit if this is removed from here?

Reply:
This all will be resolved with LocaleProvider interface.
Fist thing was fixing up the large amount of confusion between language and locale.
Next address the large amount of confusion and problems from java.util.Locale supporting some flavor of iso-630.  So, make a Locale which accepts old and new locale "code" format and delivers new IETF standard (with hyphens).  ie - you can set a LocaleProvider with underscore hyphen without countries etc .. and it will try to figure out what you want, but it will "always" provide IETF standard.

A LocalProvider can supply its own set of supported Locales.

Reply:
Not done with this - we both agree it originates in Java-Land, but the Locales need to work in IETF / Javascript land

Reply:
![image](https://user-images.githubusercontent.com/1266144/74157732-a06b8000-4bcd-11ea-8718-0ec5f7e13222.png)
Normalized all chaos to single set of Locales supplied by Java-land WebKitSpeechRecognizer.